logserver/src/test/java/alluxio/logserver/AlluxioLog4jSocketNodeTest.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.log4j.pattern.LogEvent;
+import org.apache.log4j.spi.LocationInfo;
+import org.apache.log4j.spi.LoggingEvent;
+import org.apache.log4j.spi.ThrowableInformation;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InvalidClassException;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+
+public class AlluxioLog4jSocketNodeTest {
+
+  private byte[] mBuffer;
+  private ByteArrayOutputStream mByteArrayOutputStream;
+  private ObjectOutputStream mObjectOutputStream;
+  private ValidatingObjectInputStream mValidatingObjectInputStream;
+
+  /**
+   * Only test the class in java.lang which can be serialized. Boolean, Byte, Character, Double,
+   * Float, Integer, Long, Short, String
+   */
+  private List<Object> createPositiveObjectList() {
+    List<Object> positiveObjectList = new ArrayList<>();
+    Logger logger = Logger.getLogger("a");
+    positiveObjectList.add(new Hashtable<>());
+    positiveObjectList.add(
+        new LoggingEvent(
+            "fqnOfCategoryClass",
+            logger,
+            Level.DEBUG,
+            "message",
+            new Throwable()));
+    positiveObjectList.add(new LocationInfo("c", "b", "c", "d"));
+    positiveObjectList.add(new ThrowableInformation(new Throwable()));
+    positiveObjectList.add(true);
+    positiveObjectList.add((byte) 0x11);
+    positiveObjectList.add(0.0);
+    positiveObjectList.add(1.0F);
+    positiveObjectList.add(1);
+    positiveObjectList.add(1L);
+    positiveObjectList.add((short) 1);
+    positiveObjectList.add("string");
+    return positiveObjectList;
+  }
+
+  private List<Object> createNegativeObjectList() {
+    List<Object> positiveObjectList = new ArrayList<>();
+    Logger logger = Logger.getLogger("a");
+    positiveObjectList.add(new LogEvent(
+        "category",
+        logger,
+        Level.DEBUG,
+        "message",
+        new Throwable()));
+    positiveObjectList.add(new LogLevel("lable", 1));
+    positiveObjectList.add(new AdapterLogRecord());
+    positiveObjectList.add(new Log4JLogRecord());
+    positiveObjectList.add(new LogTableColumn("label"));
+    positiveObjectList.add(Level.DEBUG);
+    positiveObjectList.add(UtilLoggingLevel.INFO);
+    positiveObjectList.add(new Throwable());
+    positiveObjectList.add(new LogLevelFormatException("message"));
+    positiveObjectList.add(new LogTableColumnFormatException("message"));
+    positiveObjectList.add(new PropertySetterException("string"));
+    return positiveObjectList;
+  }
+
+  private void createOutputStreams() {
+    try {
+      mByteArrayOutputStream = new ByteArrayOutputStream();
+      mObjectOutputStream = new ObjectOutputStream(mByteArrayOutputStream);
+    } catch (IOException exception) {
+      System.out.println(exception.getMessage());
+    }
+  }
```

Comment:
Inline this method, and replace with try-with-resources construct: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html